### PR TITLE
Handle context done in mock management plane subscribe function

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

- Handle context done in mock management plane subscribe function
- Clean up subscriptions in grpc client plugin

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

**Review by craigell — APPROVED** [review]

File: test/mock/grpc/mock_management_command_service.go [hunk 1]
```diff
@@ -136,8 +136,34 @@ func (cs *CommandService) UpdateDataPlaneHealth(
...
+	ctx := in.Context()
+
+	go cs.listenForDataPlaneResponses(ctx, in)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			slog.Info("Starting Subscribe")
```
Comment (olli-holmala) [inline test/mock/grpc/mock_management_command_service.go:148]:
```suggestion
			slog.InfoContext(ctx, "Starting Subscribe")
```

File: test/mock/grpc/mock_management_command_service.go [hunk 2]
```diff
@@ -136,8 +136,34 @@ func (cs *CommandService) UpdateDataPlaneHealth(
...
+			request := <-cs.requestChan
+
+			slog.Debug("Subscribe", "request", request)
```
Comment (olli-holmala) [inline test/mock/grpc/mock_management_command_service.go:151]:
```suggestion
			slog.DebugContext(ctx, "Subscribe", "request", request)
```

File: test/mock/grpc/mock_management_command_service.go (hunk 1 above)
Comment (olli-holmala) [reply test/mock/grpc/mock_management_command_service.go:148]: Perhaps move this entry before the for loop, so we don't spam it in the logs.

File: test/mock/grpc/mock_management_command_service.go [hunk 3]
```diff
@@ -136,8 +136,34 @@ func (cs *CommandService) UpdateDataPlaneHealth(
...
+	}
+}
+
+func (cs *CommandService) listenForDataPlaneResponses(ctx context.Context, in v1.CommandService_SubscribeServer) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
 			dataPlaneResponse, err := in.Recv()
```
Comment (olli-holmala) [inline test/mock/grpc/mock_management_command_service.go:167]: I realized something kinda fundamental, have actually defined the receive and send streams the wrong way around in our Protos?
```
// A decoupled communication mechanism between the data plane and management plane.
// buf:lint:ignore RPC_RESPONSE_STANDARD_NAME
// buf:lint:ignore RPC_REQUEST_STANDARD_NAME
rpc Subscribe(stream DataPlaneResponse) returns (stream ManagementPlaneRequest) {}
```
We should be receiving ManagementPlaneRequests and sending DataPlaneResponses from the subscription (`v1.CommandService_SubscribeServer` in this case) but instead in our code it's vice versa.

Comment (olli-holmala) [reply test/mock/grpc/mock_management_command_service.go:167]: As @aphralG correctly pointed out, this is the mock management plane, so it's Subscribe is naturally gonna be the reverse of the Agent's:
```go
type CommandService_SubscribeServer interface {
	Send(*ManagementPlaneRequest) error
	Recv() (*DataPlaneResponse, error)
	grpc.ServerStream
}
```
Resolving this comment and grabbing more coffee! ☕

**Review by olli-holmala — APPROVED** [review]

**Review by Dean-Coakley — APPROVED** [review]

**Pull request merged**